### PR TITLE
[vcpkg baseline][vcpkg-build-make] Use the same MSYS_ROOT in configure and build

### LINKED ISSUE #19337: [gettext] build failure
**Host Environment**
- OS: Windows 10 Build 19041.110
- Compiler: MSVC 2019 19.29.30038.1

**To Reproduce**
```
git clone https://github.com/microsoft/vcpkg.git
.\vcpkg\bootstrap-vcpkg.bat
.\vcpkg\vcpkg.exe install gettext
```

**Failure logs** 
```
PS C:\src> git clone https://github.com/microsoft/vcpkg.git
Cloning into 'vcpkg'...
remote: Enumerating objects: 116935, done.
remote: Counting objects: 100% (486/486), done.
remote: Compressing objects: 100% (236/236), done.
remote: Total 116935 (delta 288), reused 387 (delta 250), pack-reused 116449
Receiving objects: 100% (116935/116935), 37.07 MiB | 12.05 MiB/s, done.
Resolving deltas: 100% (72618/72618), done.
Checking connectivity... done.
Checking out files: 100% (7897/7897), done.
PS C:\src> .\vcpkg\bootstrap-vcpkg.bat
Downloading https://github.com/microsoft/vcpkg-tool/releases/download/2021-07-26/vcpkg.exe -> C:\src\vcpkg\vcpkg.exe
Done.

Telemetry
---------
vcpkg collects usage data in order to help us improve your experience.
The data collected by Microsoft is anonymous.
You can opt-out of telemetry by re-running the bootstrap-vcpkg script with -disableMetrics,
passing --disable-metrics to vcpkg on the command line,
or by setting the VCPKG_DISABLE_METRICS environment variable.

Read more about vcpkg telemetry at docs/about/privacy.md
PS C:\src> .\vcpkg\vcpkg.exe install gettext
Computing installation plan...
A suitable version of cmake was not found (required v3.20.2). Downloading portable cmake v3.20.2...
Downloading cmake...
  https://github.com/Kitware/CMake/releases/download/v3.20.2/cmake-3.20.2-windows-i386.zip -> C:\src\vcpkg\downloads\cmake-3.20.2-windows-i386.zip
Extracting cmake...
A suitable version of 7zip was not found (required v18.1.0). Downloading portable 7zip v18.1.0...
Downloading 7zip...
  https://www.nuget.org/api/v2/package/7-Zip.CommandLine/18.1.0 -> C:\src\vcpkg\downloads\7-zip.commandline.18.1.0.nupkg
Extracting 7zip...
A suitable version of nuget was not found (required v5.10.0). Downloading portable nuget v5.10.0...
Downloading nuget...
  https://dist.nuget.org/win-x86-commandline/v5.10.0/nuget.exe -> C:\src\vcpkg\downloads\c4b7375e-nuget.exe
The following packages will be built and installed:
    gettext[core]:x86-windows -> 0.21#4
  * libiconv[core]:x86-windows -> 1.16#9
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x86-windows...
A suitable version of git was not found (required v2.32.0). Downloading portable git v2.32.0...
Downloading git...
  https://github.com/git-for-windows/git/releases/download/v2.32.0.windows.2/PortableGit-2.32.0.2-32-bit.7z.exe -> C:\src\vcpkg\downloads\PortableGit-2.32.0.2-32-bit.7z.exe
Extracting git...
A suitable version of powershell-core was not found (required v7.1.3). Downloading portable powershell-core v7.1.3...
Downloading powershell-core...
  https://github.com/PowerShell/PowerShell/releases/download/v7.1.3/PowerShell-7.1.3-win-x86.zip -> C:\src\vcpkg\downloads\PowerShell-7.1.3-win-x86.zip
Extracting powershell-core...
Could not locate cached archive: C:\Users\schaical\AppData\Local\vcpkg\archives\f1\f131771992b28c24e631cba861094e96b3e9a30f52f31f6ac67319f4cd94b85c.zip
Could not locate cached archive: C:\Users\schaical\AppData\Local\vcpkg\archives\91\91b29c33f8a9e68817f8205aa7169c6293f0a4104b6db6fe60871ddfb7d33595.zip
Starting package 1/2: libiconv:x86-windows
Building package libiconv[core]:x86-windows...
-- Downloading https://ftp.gnu.org/gnu/libiconv/libiconv-1.16.tar.gz;https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/libiconv/libiconv-1.16.tar.gz -> libiconv-1.16.tar.gz...
-- Extracting source C:/src/vcpkg/downloads/libiconv-1.16.tar.gz
-- Applying patch 0002-Config-for-MSVC.patch
-- Applying patch 0003-Add-export.patch
-- Applying patch 0004-ModuleFileName.patch
-- Using source at C:/src/vcpkg/buildtrees/libiconv/src/1.16-b0c41d3469.clean
-- Found external ninja('1.10.2').
-- Getting CMake variables for x86-windows
-- Using msys root at C:/src/vcpkg/downloads/tools/msys2/703e5aa15d402568
-- Using msys root at C:/src/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8
-- Configuring x86-windows-dbg
-- Configuring x86-windows-rel
-- Using msys root at C:/src/vcpkg/downloads/tools/msys2/aaafe87dcead272e
-- Building x86-windows-dbg
-- Installing x86-windows-dbg
-- Building x86-windows-rel
-- Installing x86-windows-rel
CMake Warning at scripts/cmake/vcpkg_copy_pdbs.cmake:70 (message):
  Could not find a matching pdb file for:

      C:/src/vcpkg/packages/libiconv_x86-windows/bin/charset-1.dll
      C:/src/vcpkg/packages/libiconv_x86-windows/bin/iconv-2.dll
      C:/src/vcpkg/packages/libiconv_x86-windows/debug/bin/charset-1.dll
      C:/src/vcpkg/packages/libiconv_x86-windows/debug/bin/iconv-2.dll

Call Stack (most recent call first):
  ports/libiconv/portfile.cmake:40 (vcpkg_copy_pdbs)
  scripts/ports.cmake:141 (include)


-- Installing: C:/src/vcpkg/packages/libiconv_x86-windows/share/libiconv/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package libiconv[core]:x86-windows... done
Installing package libiconv[core]:x86-windows...
Installing package libiconv[core]:x86-windows... done
Elapsed time for package libiconv:x86-windows: 8.312 min
Starting package 2/2: gettext:x86-windows
Building package gettext[core]:x86-windows...
-- Downloading https://ftp.gnu.org/pub/gnu/gettext/gettext-0.21.tar.gz;https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/gettext/gettext-0.21.tar.gz -> gettext-0.21.tar.gz...
-- Extracting source C:/src/vcpkg/downloads/gettext-0.21.tar.gz
-- Applying patch 0002-Fix-uwp-build.patch
-- Applying patch 0003-Fix-win-unicode-paths.patch
-- Applying patch rel_path.patch
-- Applying patch android.patch
-- Using source at C:/src/vcpkg/buildtrees/gettext/src/0.21-d3cbdc875d.clean
-- Downloading https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=cfhcable;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=pilotfiber;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=gigenet;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=versaweb;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=ayera;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=netactuate;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=phoenixnap;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=astuteinternet;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=freefr;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=netcologne;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=deac-riga;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=excellmedia;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=iweb;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=jaist;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=jztkft;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=managedway;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=nchc;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=netix;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=ufpr;https://sourceforge.net/projects/winflexbison//files/winflexbison-2.5.16.zip/download?use_mirror=tenet -> winflexbison-2.5.16.zip...
-- Extracting source C:/src/vcpkg/downloads/winflexbison-2.5.16.zip
-- Using source at C:/src/vcpkg/downloads/tools/winflexbison/0a14154bff-a8cf65db07
-- Found external ninja('1.10.2').
-- Getting CMake variables for x86-windows
-- Using msys root at C:/src/vcpkg/downloads/tools/msys2/703e5aa15d402568
-- Using msys root at C:/src/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8
-- Configuring x86-windows-dbg
-- Configuring x86-windows-rel
-- Using msys root at C:/src/vcpkg/downloads/tools/msys2/aaafe87dcead272e
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_build_make.cmake:201 (message):
  libtool could not find a file being linked against!
Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_make.cmake:26 (vcpkg_build_make)
  ports/gettext/portfile.cmake:96 (vcpkg_install_make)
  ports/gettext/portfile.cmake:138 (build_libintl_only)
  scripts/ports.cmake:141 (include)


Error: Building package gettext:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: gettext:x86-windows
  Vcpkg version: 2021-07-26-9425cf5f512f242c0bcbabac31f08832825aee81

Additionally, attach any relevant sections from the log files above.
```

[build-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6924915/build-x86-windows-dbg-out.log)
[cmake-vars-x86-windows-rel.cmake.log](https://github.com/microsoft/vcpkg/files/6924916/cmake-vars-x86-windows-rel.cmake.log)
[cmake-vars-x86-windows.cmake.log](https://github.com/microsoft/vcpkg/files/6924917/cmake-vars-x86-windows.cmake.log)
[config-x86-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/6924919/config-x86-windows-rel-out.log)
[config.log-x86-windows-rel.log](https://github.com/microsoft/vcpkg/files/6924920/config.log-x86-windows-rel.log)
[get-cmake-vars-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/6924921/get-cmake-vars-x86-windows-out.log)
[stdout-x86-windows.log](https://github.com/microsoft/vcpkg/files/6924922/stdout-x86-windows.log)

### LINKED ISSUE #19353: [mpfr] x64-windows build failure
**Host Environment**
- OS: Windows 10
- Compiler: Visual Studio 2019

**To Reproduce**
`vcpkg install mpfr`

**Failure logs** 
```
C:\dev\vcpkg>vcpkg install mpfr
Computing installation plan...
The following packages will be built and installed:
    mpfr[core]:x64-windows -> 4.1.0
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\jasju\AppData\Local\vcpkg\archives\f0\f0d516b8b3d8dcbae6abfb8113289a1423199153e1d1602a9221b67f9c638f63.zip
Starting package 1/1: mpfr:x64-windows
Building package mpfr[core]:x64-windows...
-- Using C:/dev/vcpkg/downloads/mpfr-4.1.0.tar.xz
-- Cleaning sources at C:/dev/vcpkg/buildtrees/mpfr/src/mpfr-4-8955d4bc66.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/dev/vcpkg/downloads/mpfr-4.1.0.tar.xz
-- Applying patch gmpd.patch
-- Applying patch dll.patch
-- Using source at C:/dev/vcpkg/buildtrees/mpfr/src/mpfr-4-8955d4bc66.clean
-- Getting CMake variables for x64-windows
-- Using msys root at C:/dev/vcpkg/downloads/tools/msys2/41652b92f1636a16
-- Using msys root at C:/dev/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8
-- Generating configure for x64-windows
-- Finished generating configure for x64-windows
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Using msys root at C:/dev/vcpkg/downloads/tools/msys2/aaafe87dcead272e
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:155 (message):
    Command failed: C:/dev/vcpkg/downloads/tools/msys2/aaafe87dcead272e/usr/bin/make.exe -j 9 --trace -f Makefile all
    Working Directory: C:/dev/vcpkg/buildtrees/mpfr/x64-windows-dbg
    See logs for more information:
      C:\dev\vcpkg\buildtrees\mpfr\build-x64-windows-dbg-out.log
      C:\dev\vcpkg\buildtrees\mpfr\build-x64-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:191 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:26 (vcpkg_build_make)
  ports/mpfr/portfile.cmake:23 (vcpkg_install_make)
  scripts/ports.cmake:141 (include)


Error: Building package mpfr:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: mpfr:x64-windows
  Vcpkg version: 2021-07-26-9425cf5f512f242c0bcbabac31f08832825aee81

Additionally, attach any relevant sections from the log files above.
```

build-x64-windows-dbg-err.log
```
/c/dev/vcpkg/buildtrees/mpfr/src/mpfr-4-8955d4bc66.clean/missing: /c/dev/vcpkg/downloads/tools/msys2/41652b92f1636a16/usr/bin/makeinfo: /usr/bin/perl: bad interpreter: No such file or directory
make[1]: *** [Makefile:379: .././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.info] Error 126
make: *** [Makefile:499: all-recursive] Error 1
```

build-x64-windows-dbg-out.log
```
Makefile:499: target 'all-recursive' does not exist
fail=; \
if (target_option=k; case ${target_option-} in ?) ;; *) echo "am__make_running_with_option: internal error: invalid" "target option '${target_option-}' specified" >&2; exit 1;; esac; has_opt=no; sane_makeflags=$MAKEFLAGS; if { if test -z '0'; then false; elif test -n 'x86_64-pc-msys'; then true; elif test -n '4.3' && test -n '/c/dev/vcpkg/buildtrees/mpfr/x64-windows-dbg'; then true; else false; fi; }; then sane_makeflags=$MFLAGS; else case $MAKEFLAGS in *\\[\ \	]*) bs=\\; sane_makeflags=`printf '%s\n' "$MAKEFLAGS" | sed "s/$bs$bs[$bs $bs	]*//g"`;; esac; fi; skip_next=no; strip_trailopt () { flg=`printf '%s\n' "$flg" | sed "s/$1.*$//"`; }; for flg in $sane_makeflags; do test $skip_next = yes && { skip_next=no; continue; }; case $flg in *=*|--*) continue;; -*I) strip_trailopt 'I'; skip_next=yes;; -*I?*) strip_trailopt 'I';; -*O) strip_trailopt 'O'; skip_next=yes;; -*O?*) strip_trailopt 'O';; -*l) strip_trailopt 'l'; skip_next=yes;; -*l?*) strip_trailopt 'l';; -[dEDm]) skip_next=yes;; -[JT]) skip_next=yes;; esac; case $flg in *$target_option*) has_opt=yes; break;; esac; done; test $has_opt = yes); then \
  failcom='fail=yes'; \
else \
  failcom='exit 1'; \
fi; \
dot_seen=no; \
target=`echo all-recursive | sed s/-recursive//`; \
case "all-recursive" in \
  distclean-* | maintainer-clean-*) list='doc src tests tune tools/bench' ;; \
  *) list='doc src tests tune tools/bench' ;; \
esac; \
for subdir in $list; do \
  echo "Making $target in $subdir"; \
  if test "$subdir" = "."; then \
    dot_seen=yes; \
    local_target="$target-am"; \
  else \
    local_target="$target"; \
  fi; \
  (CDPATH="${ZSH_VERSION+.}:" && cd $subdir && /usr/bin/make  $local_target) \
  || eval $failcom; \
done; \
if test "$dot_seen" = "no"; then \
  /usr/bin/make  "$target-am" || exit 1; \
fi; test -z "$fail"
Making all in doc
make[1]: Entering directory '/c/dev/vcpkg/buildtrees/mpfr/x64-windows-dbg/doc'
Makefile:379: update target '.././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.info' due to: .././../src/mpfr-4-8955d4bc66.clean/doc/texinfo.tex
restore=: && backupdir=".am$$" && \
am__cwd=`pwd` && CDPATH="${ZSH_VERSION+.}:" && cd .././../src/mpfr-4-8955d4bc66.clean/doc && \
rm -rf $backupdir && mkdir $backupdir && \
if (/bin/sh /c/dev/vcpkg/buildtrees/mpfr/src/mpfr-4-8955d4bc66.clean/missing makeinfo --version) >/dev/null 2>&1; then \
  for f in .././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.info .././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.info-[0-9] .././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.info-[0-9][0-9] .././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.i[0-9] .././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.i[0-9][0-9]; do \
    if test -f $f; then mv $f $backupdir; restore=mv; else :; fi; \
  done; \
else :; fi && \
cd "$am__cwd"; \
if /bin/sh /c/dev/vcpkg/buildtrees/mpfr/src/mpfr-4-8955d4bc66.clean/missing makeinfo  --enable-encoding -I .././../src/mpfr-4-8955d4bc66.clean/doc \
 -o .././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.info .././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.texi; \
then \
  rc=0; \
  CDPATH="${ZSH_VERSION+.}:" && cd .././../src/mpfr-4-8955d4bc66.clean/doc; \
else \
  rc=$?; \
  CDPATH="${ZSH_VERSION+.}:" && cd .././../src/mpfr-4-8955d4bc66.clean/doc && \
  $restore $backupdir/* `echo "./.././../src/mpfr-4-8955d4bc66.clean/doc/mpfr.info" | sed 's|[^/]*$||'`; \
fi; \
rm -rf $backupdir; exit $rc
make[1]: Leaving directory '/c/dev/vcpkg/buildtrees/mpfr/x64-windows-dbg/doc'
```

### PR DESCRIPTION
**Pull request by JackBoosY**
`vcpkg_acquire_msys(MSYS_ROOT)` in `vcpkg_build_make` will cause the extra msys components missing when building ports.
For example, building `starlink-ast:x64-uwp`:
```
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
libtool: link: warning: undefined symbols not allowed in x86_64-unknown-mingw32 shared libraries
/bin/sh: /usr/bin/perl: No such file or directory
make[1]: *** [Makefile:4510: ast.h] Error 127
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:1674: all] Error 2
```
In starlink-ast's portfile.cmake:
```cmake
vcpkg_acquire_msys(MSYS_ROOT PACKAGES make automake1.16 perl)    
vcpkg_configure_make(
...
)

vcpkg_install_make()
```
This will overwrite the value `MSYS_ROOT` and use the incorrect make.exe to build it.

This code was introduced in PR https://github.com/microsoft/vcpkg/pull/10402.

### CODE REVIEW

**Comment by JackBoosY** [comment]
cc @Neumann-A

**Comment by dg0yt** [comment]
I guess this PR will break `vcpkg_configure_make` because the msys root lacks important tools.

Every call to `vcpkg_acquire_msys()` leads to a different tree for a differen set of packages. starlink-ast needs to request perl via the `ADDITIONAL_MSYS_PACKAGES` parameter to `vcpkg_configure_make`.

**Comment by JackBoosY** [comment]
@dg0yt  That's not related to the configure step but related to the build step.

**Comment by dg0yt** [comment]
> That's not related to the configure step but related to the build step.

Indeed. And it works, due to the default packages. 

Still, this proliferates different interfaces for defining the msys environment. 
- `ADDITIONAL_MSYS_PACKAGES` parameter to `vcpkg_configure_make`
- `MSYS_ROOT` *variable* for `vcpkg_build_make`

This creates two different msys roots for a single package like starlink-ast (and a third one if acquiring pkg-config).

I don't think this is the best way forward.

**Comment by JackBoosY** [comment]
@Neumann-A what do you think about?

**Comment by JackBoosY** [comment]
> > That's not related to the configure step but related to the build step.
> 
> Indeed. And it works, due to the default packages.
> 
> Still, this proliferates different interfaces for defining the msys environment.
> 
> * `ADDITIONAL_MSYS_PACKAGES` parameter to `vcpkg_configure_make`
> * `MSYS_ROOT` _variable_ for `vcpkg_build_make`
> 
> This creates two different msys roots for a single package like starlink-ast (and a third one if acquiring pkg-config).
> 
> I don't think this is the best way forward.

Yes, the `vcpkg_acquire_msys` before `vcpkg_configure_make` needs to be deleted and the parameters should be passed to `vcpkg_configure_make`, and the same `MSYS_ROOT` is used in `vcpkg_build_make` to solve the problem of inconsistent configuration and build using MSYS2 path.

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 1]
```diff
@@ -305,7 +309,7 @@ function(vcpkg_configure_make)
     # Pre-processing windows configure requirements
     if (VCPKG_TARGET_IS_WINDOWS)
         if(CMAKE_HOST_WIN32)
-            list(APPEND MSYS_REQUIRE_PACKAGES binutils libtool autoconf automake-wrapper automake1.16 m4)
+            list(APPEND MSYS_REQUIRE_PACKAGES binutils libtool autoconf automake-wrapper automake1.16 m4 make)
```
Comment (dg0yt) [inline scripts/cmake/vcpkg_configure_make.cmake:312]: make is in the default packages:
https://github.com/microsoft/vcpkg/blob/bd5ea16b97e91cb620fed0e10b7d9b3a8a943a52/scripts/cmake/vcpkg_acquire_msys.cmake#L152-L154

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 2]
```diff
@@ -866,4 +870,7 @@ function(vcpkg_configure_make)
 
     SET(_VCPKG_PROJECT_SOURCE_PATH ${_csc_SOURCE_PATH} PARENT_SCOPE)
     set(_VCPKG_PROJECT_SUBPATH ${_csc_PROJECT_SUBPATH} PARENT_SCOPE)
+    if (MSYS_ROOT)
+        set(MSYS_ROOT ${MSYS_ROOT} PARENT_SCOPE)
```
Comment (dg0yt) [inline scripts/cmake/vcpkg_configure_make.cmake:874]: This adds the side effect of changing `MSYS_ROOT` in the parent scope without user control. I don't think this is desirable. Either the scripts interact via `Z_`-prefixed variables (it is an implementation detail), or the variable name must be taken from yet another parameter.

**Comment by dg0yt** [comment]
I implemented passing of the make tool (from MSYS_ROOT or system) from `vcpkg_configure_make` to `vcpkg_build_make` in #19361, and CI already succeeded for x64-uwp, building gettext[core]:x64-uwp, (host) gettext[core,tools]:x64-windows, and many other ports
That PR took inspiration from this PR here. The parallel work was only meant to speed up resolving the urgent make/gettext issue. Pick what seems more useful.

**Comment by JackBoosY** [comment]
Close this PR in a favor of #19361.

**Pull request closed without merging**